dom.bs

```diff
@@ -2203,14 +2214,28 @@ for a given <a>slot</a> <var>slot</var>, run these steps:</p>
 <ol>
  <li><p>Let <var>result</var> be an empty list.</p></li>
 
- <li><p>If <var>slot</var>'s <a for=tree>root</a> is not a <a for=/>shadow root</a>, then return
+ <li><p>Let <var>root</var> be <var>slot</var>'s <a for=tree>root</a>.</p></li>
+
+ <li><p>If <var>root</var> is not a <a for=/>shadow root</a>, then return
  <var>result</var>.</p></li>
 
- <li><p>Let <var>host</var> be <var>slot</var>'s <a for=tree>root</a>'s
- <a for=DocumentFragment>host</a>.</p></li>
+ <li><p>Let <var>host</var> be <var>root</var>'s <a for=DocumentFragment>host</a>.</p></li>
+
+ <li>
+  <p>If <var>root</var>'s <a for=ShadowRoot>slot assignment</a> is "<code>manual</code>",
+  then:</p>
+
+  <ol>
+   <li><p>Let <var>result</var> be an empty list.</p></li>
+
+   <li><p>For each <a>slottable</a> in <var>slot</var>'s <a>manually assigned nodes</a>,
+   if <a>slottable</a>'s <a for=tree>parent</a> is <var>host</var>, add <a>slottable</a>
+   to <var>result</var>.</p></li>
+  </ol>
+ </li>
 
  <li>
-  <p>For each <a>slottable</a> <a for=tree>child</a> of <var>host</var>, <var>slottable</var>, in
+  <p>Otherwise, for each <a>slottable</a> <a for=tree>child</a> of <var>host</var>, <var>slottable</var>, in
```

Comment:
I guess there is a question about what we're gonna do about slots which has non-empty manually assigned nodes inserted inside a shadow tree whose assignment mode is "name". Maybe clearing the manual assignment is the simplest thing to do in that case? It would be very mysterious if manually assigned nodes are completely ignored and it was treated as the default slot and/or it was silently ignored.

Reply:
Hmm, I thought we were leaving these as-is. I.e. while that slot is located in a "named" mode shadow root, then "named" mode takes precedence. But nothing, other than re-assigning a node to another slot, removes anything from manually assigned nodes. I.e. `.assign()` is the only thing with the power to add/remove from manually assigned nodes. And no amount of node movement or slot movement can cause assignments to get cleared. I kind of feel like that's the easiest mental model for developers to have, rather than having to remember never to put a manually assigned slot into the wrong shadow root. It seems like a common pattern might be:

```javascript
shadow.appendChild(slot);
slot.assign([a,b,c]);
shadow.slotAssignment = 'named';
```

WDYT?

Reply:
I don't understand the last line of your example. `slotAssignment` is readonly. I do tend to agree that `assign()` being the only way to clear seems more straightforward than having insertion in certain trees have side effects.

Reply:
Yeah sorry, good point. That's not a great example. So you'd prefer this?

```javascript
const shadow = host.attachShadow({mode: 'open'}); // "named" slot assignment
shadow.appendChild(slot);
slot.assign([a,b,c]); // Has no effect
```

I suppose I could live with this. It just feels weird that the entire point of all of this conversation was to make the node-to-slot assignments "sticky" across all kinds of tree mutations. But then this would be the one exception - you can move a node or a slot anywhere in any tree and the references will be maintained, **except** if any stop is ever within a "named"-assignment shadow root.

Given the fact that the node references are unobservable, there will definitely be cases that are a bit hard to understand for developers. E.g. assigning a node to a slot, and then making the node a grandchild (instead of direct child) of the host. We'll need to develop developer tooling to make this situation better. But I don't think adding even more magic is the answer. Again, I'm ok either way, just trying to get this right.

Reply:
Sorry, re-reading your comment @annevk, it does sound like you might agree that `.assign()` should be the only way to clear/change node assignments.

I think this is the only remaining material issue to resolve.

Reply:
Yeah, we're on the same page. @rniwa can you take another look at this?

Reply:
So `assign(~)` has no side effect and its results get simply ignored? That's a bit strange but all other options are equally strange so maybe it's okay. I guess this might be something we want to solicit some developer feedback.

Reply:
Oh no, it got confusing. That's not what Mason or I mean. That particular comment was based on a misunderstanding.

`assign()` always succeeds (and doesn't no-op or throw), as per the current HTML PR. And moving a slot around doesn't affect its manually assigned nodes, as per this PR.

And the answer to your question in this subthread OP is that manually assigned nodes would be ignored in that case as it would be quite esoteric if moving a `slot` element around is a no-op for manually assigned nodes except if you move it to a shadow tree with named slot assignment.

Reply:
Yeah, I mean that `assign(~)` will succeed but it won't have any observable side effect until the slot is moved to another shadow root with manual slot assignment along with the assigned nodes, right?

Reply:
Ah yes, indeed. 😊

If we can get developer feedback in a timely fashion it seems reasonable to block on that, but otherwise I'd be inclined to land these PRs and count on the fact that making a change for that particular scenario is without risk and can be made once developers have some more experience with the API.

Reply:
Yeah, sounds good to me.